src/openms/include/OpenMS/ANALYSIS/ID/IDScoreSwitcherAlgorithm.h

```diff
@@ -68,14 +69,60 @@ namespace OpenMS
     /// Checks if the given @p score_name is of ScoreType @p type
     bool isScoreType(const String& score_name, const ScoreType& type)
     {
+      String chopped = score_name;
+      if (chopped.hasSuffix("_score"))
+      {
+        chopped = chopped.chop(6);
+      }
       const std::set<String>& possible_types = type_to_str_[type];
-      return possible_types.find(score_name) != possible_types.end();
+      return possible_types.find(chopped) != possible_types.end();
+    }
+
+    /// Gets a ScoreType enum from a String
+    static ScoreType getScoreType(String score_type)
+    {
+      if (score_type.hasSuffix("_score"))
+      {
+        score_type = score_type.chop(6);
+      }
+      score_type.toLower();
+      score_type.erase(std::remove(score_type.begin(), score_type.end(), '-'), score_type.end());
+      score_type.erase(std::remove(score_type.begin(), score_type.end(), '_'), score_type.end());
+      score_type.erase(std::remove(score_type.begin(), score_type.end(), ' '), score_type.end());
+
+      std::map<String, ScoreType> s_to_type =
```

Comment:
do we need the other score types here as well e.g. hyperscore etc.

Reply:
hyperscore is a raw type.